andela/codepirates-ah-frontend
# #167313360 Password reset

User should be able to reset their password

#### What does this PR do?

  Add password reset feature

#### Tasks to be accomplished

  - Present user with a form to take in email
  - Perform basic input validation
  - Display response from backend with link to email
  - Open password change page on click of the reset link
  - Display password reset response

#### How to manually test this feature

  - Navigate to the reset page by clicking the forgot password link on the login page
  - Enter your valid and registered email and follow instructions
  - Upon success, try logging in with the old, then new password to confirm the password has been reset

#### Related pivotal tracker stories
  [finishes #167313360]

#### Screenshots (if appropriate)

Email input form
![image](https://user-images.githubusercontent.com/36167006/67033626-7a017080-f116-11e9-8761-23aa80b9112b.png)

mobile view
![image](https://user-images.githubusercontent.com/36167006/67418880-ed4e2b00-f5cb-11e9-99f9-9906de82fc1d.png)


email sent response
![image](https://user-images.githubusercontent.com/36167006/67033749-a9b07880-f116-11e9-90c8-eed592b010dc.png)

Mobile view
![image](https://user-images.githubusercontent.com/36167006/67418790-baa43280-f5cb-11e9-944b-15138131028b.png)

reset password page
![image](https://user-images.githubusercontent.com/36167006/67033939-0875f200-f117-11e9-8eb1-b895c3931083.png)

mobile view
![image](https://user-images.githubusercontent.com/36167006/67419107-5b92ed80-f5cc-11e9-99f2-d4d2869748e1.png)


successful reset response
![image](https://user-images.githubusercontent.com/36167006/67034155-73272d80-f117-11e9-8bc8-8df0bacc4dcc.png)

mobile view
![image](https://user-images.githubusercontent.com/36167006/67419018-33a38a00-f5cc-11e9-9411-502a1d4ae472.png)


#### Questions:
